velox/functions/sparksql/Rand.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct RandFunction {
+  static constexpr bool is_deterministic = false;
+
+  std::optional<std::mt19937> generator;
+
+  FOLLY_ALWAYS_INLINE void call(double& result) {
+    result = folly::Random::randDouble01();
+  }
+
+  FOLLY_ALWAYS_INLINE void callNullable(
+      double& result,
+      const int32_t* seed,
+      const int32_t* partitionIndex) {
+    VELOX_USER_CHECK_NOT_NULL(partitionIndex, "partitionIndex cannot be null.");
```

Comment:
Why copy-paste and not callNullable like before?

Reply:
Hi @mbasmanova, with `(int64_t*)seed` previously used to do a cast, I found the unexpected seed value is obtained.
int32_t* seed's value: 0
int64_t* seed's value (cast from int32*): 140020228816896

So I did a copy-paste to make them separate. Should I extract a common method to reduce repeated code? Thanks!